pkg/server/handler/v1alpha1/workflowrun.go

```diff
@@ -152,6 +152,18 @@ func ListWorkflowRuns(ctx context.Context, project, workflow, tenant string, que
 	return types.NewListResponse(int(size), results[query.Start:end]), nil
 }
 
+// mutateTerminatingWorkflowRuns change status of workflowRun, which is under deleting, to Terminating.
+func mutateTerminatingWorkflowRuns(wfrs []v1alpha1.WorkflowRun) []v1alpha1.WorkflowRun {
+	results := make([]v1alpha1.WorkflowRun, len(wfrs))
+	for _, wfr := range wfrs {
+		if !wfr.DeletionTimestamp.IsZero() {
```

Comment:
Why not update status when set deletion timestamp? Then will not handles this for every list request. It will affects the performances.

Reply:
Setting deletion timestamp is done by the Kubernetes when received a deleting request, we can not do it. And the time complexity of this function is O(n)，I think it will not affect much about the performance. In the other hand，the `Terminating` status is just for easy to use, I don't hope it to be a real status of WorkflowRun, and like Kubernetes, it also does not have a `Terminating` status of `Pod`.